### PR TITLE
[Pten] Gru lstm migration

### PR DESCRIPTION
**Pull request by iclementine**
### PR types

Others

### PR changes

Others

### Describe

1. Move LSTM compute related files
```text
paddle/fluid/operators/math/lstm_compute.h   =>   paddle/phi/kernels/funcs/lstm_compute.h
paddle/fluid/operators/math/lstm_compute.cc  =>  paddle/phi/kernels/funcs/lstm_compute.cc
paddle/fluid/operators/math/lstm_compute.cu  =>  paddle/phi/kernels/funcs/lstm_compute.cu
paddle/fluid/operators/math/detail/lstm_kernel.h   =>   paddle/phi/kernels/funcs/detail/lstm_kernel.h
paddle/fluid/operators/math/detail/lstm_cpu_kernel.h  =>  paddle/phi/kernels/funcs/detail/lstm_cpu_kernel.h
paddle/fluid/operators/math/detail/lstm_gpu_kernel.h  =>  paddle/phi/kernels/funcs/detail/lstm_gpu_kernel.h
```

2. Move GRU compute related files;
```text
paddle/fluid/operators/math/gru_compute.h   =>   paddle/phi/kernels/funcs/gru_compute.h
paddle/fluid/operators/math/gru_compute.cc  =>  paddle/phi/kernels/funcs/gru_compute.cc
paddle/fluid/operators/math/gru_compute.cu  =>  paddle/phi/kernels/funcs/gru_compute.cu
paddle/fluid/operators/math/detail/gru_kernel.h   =>   paddle/phi/kernels/funcs/detail/gru_kernel.h
paddle/fluid/operators/math/detail/gru_cpu_kernel.h  =>  paddle/phi/kernels/funcs/detail/gru_cpu_kernel.h
paddle/fluid/operators/math/detail/gru_gpu_kernel.h  =>  paddle/phi/kernels/funcs/detail/gru_gpu_kernel.h
```
3. Move a small Activation functions system **only** used in LSTM and GRU;
```text
paddle/fluid/operators/math/detail/activation_functions.h => paddle/phi/kernels/funcs/detail/activation_functions.h
paddle/fluid/operators/math/detail/avx_functions.cc => paddle/phi/kernels/funcs/detail/avx_functions.cc
paddle/fluid/operators/math/detail/avx_mathfun.h => paddle/phi/kernels/funcs/detail/avx_mathfun.h
```

4. Move sequence2batch related files;
```text
paddle/fluid/operators/math/sequence2batch.h  => paddle/phi/kernels/funcs/sequence2batch.h
paddle/fluid/operators/math/sequence2batch.cc  => paddle/phi/kernels/funcs/sequence2batch.cc
paddle/fluid/operators/math/sequence2batch.cu  => paddle/phi/kernels/funcs/sequence2batch.cu
```

4. update all references to symbols within them.

### CODE REVIEW

File: paddle/phi/kernels/funcs/detail/lstm_gpu_kernel.h [hunk 1]
```diff
@@ -203,25 +244,45 @@ void gpu_lstm_forward(const platform::DeviceContext& context, Op op,
...
         active_state);
   }
 }
 
 template <class T, class Op>
-void gpu_lstm_backward(const platform::DeviceContext& context, Op op,
-                       LstmMetaValue<T> value, LstmMetaGrad<T> grad,
-                       int frame_size, int batch_size, T cell_clip,
-                       ActivationType active_node, ActivationType active_gate,
+void gpu_lstm_backward(const paddle::platform::DeviceContext& context,
```
Comment (chenwhql) [inline paddle/phi/kernels/funcs/detail/lstm_gpu_kernel.h:277]: paddle::platform::DeviceContext -> phi::DeviceContext

File: paddle/phi/kernels/funcs/detail/lstm_gpu_kernel.h [hunk 2]
```diff
@@ -237,21 +298,37 @@ void gpu_lstm_backward(const platform::DeviceContext& context, Op op,
   }
 
   auto stream =
-      reinterpret_cast<const platform::CUDADeviceContext&>(context).stream();
+      reinterpret_cast<const paddle::platform::CUDADeviceContext&>(context)
```
Comment (chenwhql) [inline paddle/phi/kernels/funcs/detail/lstm_gpu_kernel.h:301]: paddle::platform::CUDADeviceContext -> phi::GPUContext

File: paddle/phi/kernels/funcs/sequence2batch.h [hunk 3]
```diff
@@ -59,32 +61,37 @@ class LoDTensor2BatchFunctor {
 
  public:
   void operator()(const DeviceContext& context,
-                  const framework::LoDTensor& lod_tensor,
-                  framework::LoDTensor* batch, bool is_cal_batch_lod,
+                  const paddle::framework::LoDTensor& lod_tensor,
```
Comment (chenwhql) [inline paddle/phi/kernels/funcs/sequence2batch.h:64]: paddle::framework::LoDTensor -> DenseTensor

**Review by chenwhql — APPROVED** [review]
LGTM，一些命名空间后续PR可以追加修改为phi下面的

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**